app/src/main/java/com/naman14/timberx/ui/bindings/LastFmImageBindings.kt

```diff
@@ -167,7 +167,7 @@ private fun fetchArtistImage(
                     is Outcome.Success -> {
                         val artistResult = outcome.data.artist ?: return@subscribeForOutcome
                         val url = artistResult.artwork.ofSize(artworkSize)
-                                .url
+                                .url.replace("300x300", "700x0")
```

Comment:
This is not the right way to do it. The size is being given above from the ArtworkSize enum.

Reply:
True, maybe only replace if the given size is Mega? As far as I know one can´t get larger images than 300x300.

Reply:
Hmm, well here's an example: https://ws.audioscrobbler.com/2.0/?method=artist.getinfo&artist=Cher&api_key=fdb3a51437d4281d4d64964d333531d4&format=json

```json
"image": [
{
"#text": "https://lastfm-img2.akamaized.net/i/u/34s/8e505ad22689fd4b71f0e9c49c122a43.png",
"size": "small"
},
{
"#text": "https://lastfm-img2.akamaized.net/i/u/64s/8e505ad22689fd4b71f0e9c49c122a43.png",
"size": "medium"
},
{
"#text": "https://lastfm-img2.akamaized.net/i/u/174s/8e505ad22689fd4b71f0e9c49c122a43.png",
"size": "large"
},
{
"#text": "https://lastfm-img2.akamaized.net/i/u/300x300/8e505ad22689fd4b71f0e9c49c122a43.png",
"size": "extralarge"
},
{
"#text": "https://lastfm-img2.akamaized.net/i/u/300x300/8e505ad22689fd4b71f0e9c49c122a43.png",
"size": "mega"
},
{
"#text": "https://lastfm-img2.akamaized.net/i/u/300x300/8e505ad22689fd4b71f0e9c49c122a43.png",
"size": ""
}
]
```

It does appear 300x300 is the maximum, anything larger would just be stretching afaik.

Reply:
But maybe not - https://lastfm-img2.akamaized.net/i/u/700x700/8e505ad22689fd4b71f0e9c49c122a43.png does look okay.

It'd be better to add an option to the `ArtworkSize` enum that we take into account when processing the URL, somehow.

Reply:
Yeah, it is the maximum, that's why I'm replacing it

![affix_20190209_143013](https://user-images.githubusercontent.com/9915479/52526698-1c14b300-2c82-11e9-8ea3-22aadadb5aa1.png)

I wouldn't say the larger one would strech, it actually fits better and does not look blurry. (300x300 left, 700x0 right)

Reply:
Still - replacing there is replacing for all sizes, even list items.

Reply:
Right, we wouldn't need to replace any other than the mega size.

Reply:
Yeah that sounds good

Reply:
To make things relatively self contained you could maybe try to insert that logic into the `ofSize` extension on `List<Artwork>` inside of `LastfmAlbum.kt`. E.g.:

```kotlin
fun List<Artwork>.ofSize(size: ArtworkSize): Artwork {
    val result = firstOrNull { it.size == size.apiValue } ?: this[(this.size - 1)]
    return if (size == ArtworkSize.MEGA) {
        result.copy(url = result.url.replace("300x300", "700x700"))
    } else {
        result
    }
}
```

Reply:
Sorry, I didn't read this. Yeah. Definitely would be better that way.